diff --git a/.gitignore b/.gitignore
index c4139e3..5e32353 100644
--- a/.gitignore
+++ b/.gitignore
@@ -4,3 +4,4 @@ Homestead.yaml
 Homestead.json
 .env
 /config/database.php
+/nbproject/private/
\ No newline at end of file
diff --git a/app/Http/Controllers/PagesController.php b/app/Http/Controllers/PagesController.php
index f09e5a8..fa95d15 100644
--- a/app/Http/Controllers/PagesController.php
+++ b/app/Http/Controllers/PagesController.php
@@ -18,9 +18,9 @@ class PagesController extends Controller
      return view('pages.about');   //
     }
     
-    public function retreat()
+    public function event()
     {
-     return view('pages.retreat');   //
+     return view('pages.event');   //
     }
     
     public function reservation()
diff --git a/app/Http/Controllers/RetreatsController.php b/app/Http/Controllers/RetreatsController.php
index 3100f9f..d3f293a 100644
--- a/app/Http/Controllers/RetreatsController.php
+++ b/app/Http/Controllers/RetreatsController.php
@@ -5,8 +5,8 @@ namespace App\Http\Controllers;
 use Illuminate\Http\Request;
 use App\Http\Requests;
 use App\Http\Controllers\Controller;
-
-
+use Illuminate\Support\Facades\Redirect;
+use Input;
 
 class RetreatsController extends Controller
 {
@@ -30,6 +30,8 @@ class RetreatsController extends Controller
      */
     public function create()
     {
+        // var_dump($retreats);
+        return view('retreats.create');   //
         //
     }
 
@@ -61,10 +63,15 @@ class RetreatsController extends Controller
      * @param  int  $id
      * @return \Illuminate\Http\Response
      */
-    public function edit($id)
-    {
-        //
-    }
+    //public function edit($id)
+    //{
+    //   $retreats = \App\Retreat::();
+    //   return view('retreats.edit',compact('retreats'));
+    //  }
+public function edit(\App\Retreat $retreats)
+    { $ret = \App\Retreat::get($retreats->id);
+       return view('retreats.edit',compact('ret'));
+      }
 
     /**
      * Update the specified resource in storage.
@@ -88,4 +95,24 @@ class RetreatsController extends Controller
     {
         //
     }
+    public function saveCreate()
+    {
+        $input = Input::all();
+        $retreat = new \App\Retreat;
+        $retreat->idnumber = $input['idnumber'];
+        $retreat->title = $input['title'];
+        $retreat->description = $input['description'];
+        $retreat->start = $input['start'];
+        $retreat->end = $input['end'];
+        $retreat->type = $input['type'];
+        $retreat->silent = $input['silent'];
+        $retreat->amount = $input['amount'];
+        $retreat->year = $input['year'];
+        $retreat->directorid = $input['directorid'];
+        $retreat->innkeeperid = $input['innkeeperid'];
+        $retreat->assistantid = $input['assistantid'];
+        $retreat->save();
+        
+    return Redirect::action('RetreatsController@index');
+    }
 }
diff --git a/app/Http/routes.php b/app/Http/routes.php
index f93bd57..5669240 100644
--- a/app/Http/routes.php
+++ b/app/Http/routes.php
@@ -12,7 +12,7 @@
 */
 
 Route::get('/',['as' => 'welcome','uses' => 'PagesController@welcome']);
-Route::get('retreat',['as' => 'retreat','uses' => 'PagesController@retreat']);
+Route::get('event',['as' => 'event','uses' => 'PagesController@event']);
 Route::get('reservation',['as' => 'reservation','uses' => 'PagesController@reservation']);
 Route::get('room',['as' => 'room','uses' => 'PagesController@room']);
 Route::get('housekeeping',['as' => 'housekeeping','uses' => 'PagesController@housekeeping']);
@@ -27,8 +27,15 @@ Route::get('support',['as' => 'support','uses' => 'PagesController@support']);
 // Route::get('support','WelcomeController@index');
 Route::get('about',['as' => 'about','uses' => 'PagesController@about']);
 Route::get('phpinfo','SystemController@phpinfo');
-Route::get('retreat/index',['as' => 'retreat/index','uses' => 'RetreatsController@index']);
-
+Route::resource('retreat','RetreatsController');
+// Route::get('retreat/index',['as' => 'retreat/index','uses' => 'RetreatsController@index']);
+// Route::get('retreat/create',['as' => 'retreat/create','uses' => 'RetreatsController@create']);
+// Route::post('retreat/create', ['as' => 'retreat/create','uses' => 'RetreatsController@saveCreate']);
+//Route::get('retreat/edit/{id}',['as' => 'retreat/edit/{id}','uses' => 'RetreatsController@edit']);
+// Route::get('retreat/delete',['as' => 'retreat/delete','uses' => 'RetreatsController@delete']);
+// Route::model('retreat','Retreat');
+//Route::get('retreat/edit/{retreat}', ['as' => 'retreat/edit','uses' =>'RetreatsController@edit']);
+// Route::get('/edit/{retreat}','RetreatsController@edit');
 Route::get('addretreat', function()
 {
 $retreat = new App\Retreat;
diff --git a/app/Providers/AppServiceProvider.php b/app/Providers/AppServiceProvider.php
index 35471f6..ed3227e 100644
--- a/app/Providers/AppServiceProvider.php
+++ b/app/Providers/AppServiceProvider.php
@@ -23,6 +23,8 @@ class AppServiceProvider extends ServiceProvider
      */
     public function register()
     {
-        //
+          if ($this->app->environment() == 'local') {
+        $this->app->register('Laracasts\Generators\GeneratorsServiceProvider');
+    } //
     }
 }
diff --git a/app/Retreat.php b/app/Retreat.php
index 25e37a3..de877c0 100644
--- a/app/Retreat.php
+++ b/app/Retreat.php
@@ -3,8 +3,16 @@
 namespace App;
 
 use Illuminate\Database\Eloquent\Model;
+use Carbon\Carbon;
 
 class Retreat extends Model
 {
+    
   protected $dates = ['start', 'end', 'created_at', 'updated_at', 'disabled_at'];  //
+  public function setStartAttribute($date) {
+      $this->attributes['start'] = Carbon::parse($date);
+  }
+  public function setEndAttribute($date) {
+      $this->attributes['end'] = Carbon::parse($date);
+  }
 }
diff --git a/composer.json b/composer.json
index 4d439a7..2eef2e2 100644
--- a/composer.json
+++ b/composer.json
@@ -13,7 +13,8 @@
         "fzaninotto/faker": "~1.4",
         "mockery/mockery": "0.9.*",
         "phpunit/phpunit": "~4.0",
-        "phpspec/phpspec": "~2.1"
+        "phpspec/phpspec": "~2.1",
+        "laracasts/generators": "^1.1"
     },
     "autoload": {
         "classmap": [
diff --git a/composer.lock b/composer.lock
index c65851d..0a01638 100644
--- a/composer.lock
+++ b/composer.lock
@@ -4,7 +4,7 @@
         "Read more about it at https://getcomposer.org/doc/01-basic-usage.md#composer-lock-the-lock-file",
         "This file is @generated automatically"
     ],
-    "hash": "c18dc4c78fa43a9f42fdcd6aae53392e",
+    "hash": "fc504a2f99889e0fb56bde03a00f3d8a",
     "packages": [
         {
             "name": "classpreloader/classpreloader",
@@ -1919,6 +1919,50 @@
             "time": "2015-05-11 14:41:42"
         },
         {
+            "name": "laracasts/generators",
+            "version": "1.1.2",
+            "source": {
+                "type": "git",
+                "url": "https://github.com/laracasts/Laravel-5-Generators-Extended.git",
+                "reference": "eb58b86f49d69eff4db4cc6f2ec94480a57c4358"
+            },
+            "dist": {
+                "type": "zip",
+                "url": "https://api.github.com/repos/laracasts/Laravel-5-Generators-Extended/zipball/eb58b86f49d69eff4db4cc6f2ec94480a57c4358",
+                "reference": "eb58b86f49d69eff4db4cc6f2ec94480a57c4358",
+                "shasum": ""
+            },
+            "require": {
+                "illuminate/support": "~5.0",
+                "php": ">=5.4.0"
+            },
+            "require-dev": {
+                "phpspec/phpspec": "~2.1"
+            },
+            "type": "library",
+            "autoload": {
+                "psr-4": {
+                    "Laracasts\\Generators\\": "src/"
+                }
+            },
+            "notification-url": "https://packagist.org/downloads/",
+            "license": [
+                "MIT"
+            ],
+            "authors": [
+                {
+                    "name": "Jeffrey Way",
+                    "email": "jeffrey@laracasts.com"
+                }
+            ],
+            "description": "Extend Laravel 5's generators.",
+            "keywords": [
+                "generators",
+                "laravel"
+            ],
+            "time": "2015-08-19 14:29:06"
+        },
+        {
             "name": "mockery/mockery",
             "version": "0.9.4",
             "source": {
diff --git a/config/app.php b/config/app.php
index 7d459fe..6ccb256 100644
--- a/config/app.php
+++ b/config/app.php
@@ -26,7 +26,7 @@ return [
     |
     */
 
-    'url' => 'http://localhost',
+    'url' => 'http://localhost/montserrat/public',
 
     /*
     |--------------------------------------------------------------------------
diff --git a/database/migrations/2015_11_04_021155_create_retreats.php b/database/migrations/2015_11_04_021155_create_retreats.php
index 02f91de..a7c071b 100644
--- a/database/migrations/2015_11_04_021155_create_retreats.php
+++ b/database/migrations/2015_11_04_021155_create_retreats.php
@@ -18,8 +18,8 @@ class CreateRetreats extends Migration
             $table->integer('idnumber');
             $table->string('title');
             $table->string('description');
-            $table->bigInteger('start');
-            $table->bigInteger('end');
+            $table->timestamp('start');
+            $table->timestamp('end');
             $table->string('type');
             $table->boolean('silent');
             $table->decimal('amount',6,2);
diff --git a/resources/views/pages/event.blade.php b/resources/views/pages/event.blade.php
new file mode 100644
index 0000000..276aa29
--- /dev/null
+++ b/resources/views/pages/event.blade.php
@@ -0,0 +1,12 @@
+@extends('template')
+@section('content')
+<h1>Welcome to the Montserrat Retreat House Database!</h1>
+<p>This is the retreat page. You will be able to see a list of the retreats, create a retreat, update, a retreat and delete a retreat from this page.</p>
+<p>
+    <a href={{ action('RetreatsController@index') }}>{!! Html::image('img/index.png', 'Index of Retreats',array('title'=>"Retreat Index")) !!}</a></li>
+    <a href={{ action('RetreatsController@create') }}>{!! Html::image('img/create.png', 'Create a Retreat',array('title'=>"Create Retreat")) !!}</a></li>
+    <a href={{ action('RetreatsController@edit') }}>{!! Html::image('img/update.png', 'Update a Retreat',array('title'=>"Update Index")) !!}</a></li>
+    <a href={{ action('RetreatsController@destroy') }}>{!! Html::image('img/delete.png', 'Delete a Retreat',array('title'=>"Delete Retreat")) !!}</a></li>
+</p>
+
+@stop
diff --git a/resources/views/pages/retreat.blade.php b/resources/views/pages/retreat.blade.php
deleted file mode 100644
index e095750..0000000
--- a/resources/views/pages/retreat.blade.php
+++ /dev/null
@@ -1,12 +0,0 @@
-@extends('template')
-@section('content')
-<h1>Welcome to the Montserrat Retreat House Database!</h1>
-<p>This is the retreat page. You will be able to see a list of the retreats, create a retreat, update, a retreat and delete a retreat from this page.</p>
-<p>
-    <a href={{ route('retreat/index') }}>{!! Html::image('img/index.png', 'Index of Retreats',array('title'=>"Retreat Index")) !!}</a></li>
-    <a href={{ route('retreat/index') }}>{!! Html::image('img/create.png', 'Create a Retreat',array('title'=>"Create Retreat")) !!}</a></li>
-    <a href={{ route('retreat/index') }}>{!! Html::image('img/update.png', 'Update a Retreat',array('title'=>"Update Index")) !!}</a></li>
-    <a href={{ route('retreat/index') }}>{!! Html::image('img/delete.png', 'Delete a Retreat',array('title'=>"Delete Retreat")) !!}</a></li>
-</p>
-
-@stop
diff --git a/resources/views/retreats/create.blade.php b/resources/views/retreats/create.blade.php
new file mode 100644
index 0000000..71b6183
--- /dev/null
+++ b/resources/views/retreats/create.blade.php
@@ -0,0 +1,64 @@
+@extends('template')
+@section('content')
+
+    <section class="section-padding">
+        <div class="jumbotron text-left">
+            <h2><strong>Add A Retreat</strong></h2>
+            {!! Form::open(['url'=> 'retreat/create', 'class' => 'form']) !!}
+            <div class="form-group">
+                {!! Form::label('id', 'ID#: ') !!}
+                {!! Form::text('idnumber', null, ['class'=>'form-control']) !!}
+            </div>
+            <div class="form-group">
+                {!! Form::label('title', 'Title:') !!}
+                {!! Form::text('title', null, ['class'=>'form-control']) !!}
+            </div>
+            <div class="form-group">
+                {!! Form::label('description', 'Description:') !!}
+                {!! Form::textarea('description', null, ['class'=>'form-control']) !!}
+            </div>
+            <div class="form-group">
+                {!! Form::label('start', 'Start date: (12-25-2015)') !!}
+                {!! Form::input('date', 'start', date('Y-m-d'), ['class'=>'form-control']) !!}
+            </div>
+            <div class="form-group">
+                {!! Form::label('end', 'End date: (2015-12-11)') !!}
+                {!! Form::input('date', 'end', Carbon\Carbon::now()->addDays(3)->format('Y-m-d'), ['class'=>'form-control']) !!}
+            </div>
+            <div class="form-group">
+                {!! Form::label('type', 'Type: (Men, Women, Open, Couples, Other)') !!}
+                {!! Form::text('type', null, ['class'=>'form-control']) !!}
+            </div>
+            <div class="form-group">
+                {!! Form::label('silent', 'Silent:') !!}
+                {!! Form::text('silent', 1, ['class'=>'form-control']) !!}
+            </div>
+            <div class="form-group">
+                {!! Form::label('amount', 'Amount:') !!}
+                {!! Form::text('amount', 390, ['class'=>'form-control']) !!}
+            </div>
+            <div class="form-group">
+                {!! Form::label('year', 'Year:') !!}
+                {!! Form::text('year', 2015, ['class'=>'form-control']) !!}
+            </div>
+            <div class="form-group">
+                {!! Form::label('directorid', 'Director ID:') !!}
+                {!! Form::text('directorid', null, ['class'=>'form-control']) !!}
+            </div>
+            <div class="form-group">
+                {!! Form::label('innkeeperid', 'Innkeeper ID:') !!}
+                {!! Form::text('innkeeperid', null, ['class'=>'form-control']) !!}
+            </div>
+            <div class="form-group">
+                {!! Form::label('assistantid', 'Assistant ID:') !!}
+                {!! Form::text('assistantid', null, ['class'=>'form-control']) !!}
+            </div>
+            
+            <div class="form-group">
+                {!! Form::submit('Add Retreat', ['class'=>'btn btn-primary']) !!}
+            </div>
+                {!! Form::close() !!}
+        </div>
+    </section>
+
+@stop
\ No newline at end of file
diff --git a/resources/views/retreats/delete.blade.php b/resources/views/retreats/delete.blade.php
new file mode 100644
index 0000000..14cd46c
--- /dev/null
+++ b/resources/views/retreats/delete.blade.php
@@ -0,0 +1,41 @@
+@extends('template')
+@section('content')
+
+    <section class="section-padding">
+        <div class="jumbotron text-left">
+            <div class="panel panel-default">
+                <div class="panel-heading">
+                    <h1>
+                    <span class="grey">MRH Retreat Index</span> 
+                    </h1>
+                </div>
+                @if ($retreats->isEmpty())
+                    <p> Currently, there are no retreats!</p>
+                @else
+                <table class="table">
+                    <thead>
+                        <tr>
+                            <th>ID#</th>
+                            <th>Title</th>
+                            <th>Starts</th>
+                            <th>Ends</th>
+                            <th>Silent</th>
+                        </tr>
+                    </thead>
+                    <tbody>
+                        @foreach($retreats as $retreat)
+                        <tr>
+                            <td>{{ $retreat->idnumber}}</td>
+                            <td>{{ $retreat->title }}</td>
+                            <td>{{ date('F d, Y', strtotime($retreat->start)) }}</td>
+                            <td>{{ date('F d, Y', strtotime($retreat->end)) }}</td>
+                            <td>{{ $retreat->silent ? 'Yes' : 'No'}}</td>
+                        </tr>
+                        @endforeach
+                    </tbody>
+                </table>
+                @endif
+            </div>
+        </div>
+    </section>
+@stop
\ No newline at end of file
diff --git a/resources/views/retreats/edit.blade.php b/resources/views/retreats/edit.blade.php
new file mode 100644
index 0000000..5dcb4e6
--- /dev/null
+++ b/resources/views/retreats/edit.blade.php
@@ -0,0 +1,41 @@
+@extends('template')
+@section('content')
+
+    <section class="section-padding">
+        <div class="jumbotron text-left">
+            <div class="panel panel-default">
+                <div class="panel-heading">
+                    <h1>
+                    <span class="grey">MRH Retreat Edit</span> 
+                    </h1>
+                </div>
+                @if ({{ \App\Retreat $retreats->isEmpty() }})
+                    <p> Currently, there is no retreat!</p>
+                @else
+                <table class="table">
+                    <thead>
+                        <tr>
+                            <th>ID#</th>
+                            <th>Title</th>
+                            <th>Starts</th>
+                            <th>Ends</th>
+                            <th>Silent</th>
+                        </tr>
+                    </thead>
+                    <tbody>
+                        @foreach($retreats as $retreat)
+                        <tr>
+                            <td>{{ $retreat->idnumber}}</td>
+                            <td>{{ $retreat->title }}</td>
+                            <td>{{ date('F d, Y', strtotime($retreat->start)) }}</td>
+                            <td>{{ date('F d, Y', strtotime($retreat->end)) }}</td>
+                            <td>{{ $retreat->silent ? 'Yes' : 'No'}}</td>
+                        </tr>
+                        @endforeach
+                    </tbody>
+                </table>
+                @endif
+            </div>
+        </div>
+    </section>
+@stop
\ No newline at end of file
diff --git a/resources/views/template.blade.php b/resources/views/template.blade.php
index f8e3da0..39f5b5c 100644
--- a/resources/views/template.blade.php
+++ b/resources/views/template.blade.php
@@ -3,54 +3,18 @@
     <head>
         <meta charset="UTF-8">
         <title>Montserrat Retreat House Database</title>
-
         <link rel="stylesheet" href="//netdna.bootstrapcdn.com/bootstrap/3.0.3/css/bootstrap.min.css">
         <script src="https://code.jquery.com/jquery.js"></script>
         <script src="//netdna.bootstrapcdn.com/bootstrap/3.0.3/js/bootstrap.min.js"></script>
         <link rel="stylesheet" href="{{ asset('css/style.css') }}">
-        <style>
-            html, body {
-                height: 100%;
-            }
-
-            body {
-                margin: 0;
-                padding: 0;
-                width: 100%;
-                display: table;
-                font-weight: 100;
-                font-family: 'Lato';
-                background: buttonshadow none repeat scroll 0 0;
-            }
-
-            .container {
-                text-align: center;
-                display: table-cell;
-                vertical-align: middle;
-            }
-
-            .content {
-                text-align: center;
-                display: inline-block;
-            }
-            .footer {
-                text-align: center;
-            }
-
-            .title {
-                font-size: 96px;
-            }
-            td {
-                align: center;
-                }
-        </style>
+        
     </head>
     
     <body>
     <header>
-           <a href={{ route('welcome') }}>{!! Html::image('img/mrhlogoblack.png','Home',array('title'=>'Home')) !!}</a>
-        
-
+        <div>
+            <a href={{ route('welcome') }}>{!! Html::image('img/mrhlogoblack.png','Home',array('title'=>'Home','class'=>'logo')) !!}</a>
+        </div>
         <nav class="navbar navbar-default" role="navigation">
         <!-- Brand and toggle get grouped for better mobile display -->
         <div class="navbar-header">
@@ -65,7 +29,7 @@
         <!-- Collect the nav links, forms, and other content for toggling -->
         <div class="collapse navbar-collapse" id="bs-example-navbar-collapse-1">
             <ul class="nav navbar-nav navbar-left">
-                <li class="active"><a href={{ route('retreat') }}>{!! Html::image('img/retreat.png', 'Retreat',array('title'=>"Retreat")) !!}</a></li>
+                <li><a href={{ route('event') }}>{!! Html::image('img/retreat.png', 'Retreat',array('title'=>"Retreat")) !!}</a></li>
                 <li><a href={{ route('reservation') }}>{!! Html::image('img/reservation.png', 'Reservation',array('title'=>"Reservation")) !!}</a></li>
                 <li><a href={{ route('room') }}>{!! Html::image('img/room.png', 'Room',array('title'=>"Room")) !!}</a></li>
                 <li><a href={{ route('housekeeping') }}>{!! Html::image('img/housekeeping.png', 'Housekeeping',array('title'=>"Housekeeping")) !!}</a></li>
