--- config/entrust.php
+++ PHP_CodeSniffer
@@ -18,7 +18,7 @@
     | the role if it is in a different namespace.
     |
     */
-    'role' => 'App\Role',
+    'role' => App\Role::class,
 
     /*
     |--------------------------------------------------------------------------
@@ -39,7 +39,7 @@
     | Update the permission if it is in a different namespace.
     |
     */
-    'permission' => 'App\Permission',
+    'permission' => App\Permission::class,
 
     /*
     |--------------------------------------------------------------------------

